packages/@webex/plugin-meetings/src/meeting/index.ts

```diff
@@ -3419,274 +3340,110 @@ export default class Meeting extends StatelessWebexPlugin {
   }
 
   /**
-   * Emits the 'media:ready' event with a local stream that consists of 1 local audio and 1 local video track
-   * @returns {undefined}
-   * @private
-   * @memberof Meeting
-   */
-  private sendLocalMediaReadyEvent() {
-    Trigger.trigger(
-      this,
-      {
-        file: 'meeting/index',
-        function: 'sendLocalMediaReadyEvent',
-      },
-      EVENT_TRIGGERS.MEDIA_READY,
-      {
-        type: EVENT_TYPES.LOCAL,
-        stream: MediaUtil.createMediaStream([
-          this.mediaProperties.audioTrack?.underlyingTrack,
-          this.mediaProperties.videoTrack?.underlyingTrack,
-        ]),
-      }
-    );
-  }
-
-  /**
-   * Sets the local audio track on the class and emits an event to the developer
-   * @param {MediaStreamTrack} rawAudioTrack
-   * @param {Boolean} emitEvent if true, a media ready event is emitted to the developer
-   * @returns {undefined}
-   * @private
-   * @memberof Meeting
+   * Stores the reference to a new microphone track, sets up the required event listeners
+   * on it, cleans up previous track, etc.
+   *
+   * @param {LocalMicrophoneTrack | null} localTrack local microphone track
+   * @returns {void}
    */
-  private setLocalAudioTrack(rawAudioTrack: MediaStreamTrack | null, emitEvent = true) {
-    if (this.isMultistream) {
-      throw new Error('this method is only supposed to be used for transcoded meetings');
-    }
-
-    if (rawAudioTrack) {
-      const settings = rawAudioTrack.getSettings();
+  private setLocalAudioTrack(localTrack?: LocalMicrophoneTrack) {
+    const oldTrack = this.mediaProperties.audioTrack;
 
-      const localMicrophoneTrack = new LocalMicrophoneTrack(
-        MediaUtil.createMediaStream([rawAudioTrack])
-      );
+    oldTrack?.off(LocalTrackEvents.Muted, this.localAudioTrackMuteStateHandler);
```

Comment:
in a normal scenario this wont happen if we have the device change logic in the local track

Reply:
if we do the webrtc-core changes to allow device changing within same local track instance, then this code that listens on the underlying track change will handle it and we need the underlying track change anyway for effects to work too